pkg/version/version.go

```diff
@@ -0,0 +1,34 @@
+//  Licensed to the Apache Software Foundation (ASF) under one or more
+//  contributor license agreements.  See the NOTICE file distributed with
+//  this work for additional information regarding copyright ownership.
+//  The ASF licenses this file to You under the Apache License, Version 2.0
+//  (the "License"); you may not use this file except in compliance with
+//  the License.  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package version
+
+import (
+	"fmt"
+	"runtime"
+)
+
+var (
+	// The following fields are populated at build time using -ldflags -X.
+	_buildVersion = "unknown"
+	_buildGitRevision = "unknown"
+
+	_buildGoVersion = runtime.Version()
+)
+
+// Version produces a single-line version info with format:
+// <version>-<git revision>-<go version>
```

Comment:
Suggest to add information about the operating system to facilitate troubleshooting due to different build platforms.

Reply:
You mean the runnning OS or the building OS?

Reply:
building OS

Reply:
OK, will add it.

Reply:
All right, i changed my mind, the build os will confuse users when the binary was compiled crossly.

I will add a verbose version info, which contains this.